docker/rust-all.Dockerfile

```diff
@@ -278,44 +317,54 @@ ENV GIT_SHA ${GIT_SHA}
 ### EXPERIMENTAL ###
 
 ### Validator Image ###
-FROM validator AS validator-testing
+# We will build a base testing image with the necessary packages and 
+# duplicate steps from validator step. This will, however, reduce 
+# cache invalidation and reduce build times. 
+FROM validator-testing-base  AS validator-testing
 
 RUN apt-get update && apt-get install -y \
-    # Extra goodies for debugging
-    less \
-    git \
-    vim \
-    nano \
-    libjemalloc-dev \
-    binutils \
-    graphviz \
-    ghostscript \
-    strace \
-    htop \
-    sysstat \
-    valgrind \
+    libssl1.1 \
+    ca-certificates \
+    # Needed to run debugging tools like perf
+    linux-perf \
```

Comment:
why not move these extra apto-get installs (for perf and others) into the `validator-testing-base` as well?

Reply:
Sure, I have moved it around now.

As far time savings, its unclear because our build is actually parallelized. But, I would expect we save up to 5 minutes because that how long bcc build/related tools takes.